egs/wsj/s5/steps/data/reverberate_data_dir.py

```diff
@@ -0,0 +1,470 @@
+                                                    )
+
+    assert len(noise_addition_descriptor['noise_io']) == len(noise_addition_descriptor['start_times'])
+    assert len(noise_addition_descriptor['noise_io']) == len(noise_addition_descriptor['snrs'])
+    if len(noise_addition_descriptor['noise_io']) > 0:
+        reverberate_opts += "--additive-signals='{0}' ".format(','.join(noise_addition_descriptor['noise_io']))
+        reverberate_opts += "--start-times='{0}' ".format(','.join(map(lambda x:str(x), noise_addition_descriptor['start_times'])))
+        reverberate_opts += "--snrs='{0}' ".format(','.join(map(lambda x:str(x), noise_addition_descriptor['snrs'])))
+
+    return reverberate_opts
+
+
+# This is the main function to generate pipeline command for the corruption
+# The generic command of wav-reverberate will be like:
+# wav-reverberate --duration=t --impulse-response=rir.wav 
+# --additive-signals='noise1.wav,noise2.wav' --snrs='snr1,snr2' --start-times='s1,s2' input.wav output.wav
+def CorruptWav(wav_scp,  # the dictionary of which elements are the IO of the speech recordings
+               durations, # the dictionary of which elements are the duration (in sec) of the speech recordings
+               output_dir, # output directory to write the corrupted wav.scp 
+               room_dict,  # the room dictionary, please refer to MakeRoomDict() for the format
+               point_noise_list, # the point source noise list
+               iso_noise_list, # the isotropic noise list
+               foreground_snr_array, # the SNR for adding the foreground noises
+               background_snr_array, # the SNR for adding the background noises
+               num_replicas, # Number of replicate to generated for the data
+               prefix, # prefix for the id of the corrupted utterances
+               speech_rvb_probability, # Probability of reverberating a speech signal
+               isotropic_noise_addition_probability, # Probability of adding isotropic noises
+               pointsource_noise_addition_probability, # Probability of adding point-source noises
+               max_noises_per_minute # maximum number of point-source noises that can be added to a recording according to its duration
+               ):
+    foreground_snrs = list_cyclic_iterator(foreground_snr_array)
+    background_snrs = list_cyclic_iterator(background_snr_array)
+    corrupted_wav_scp = {}
+    for i in range(num_replicas):
+        keys = wav_scp.keys()
+        keys.sort()
+        for wav_id in keys:
+            wav_original_pipe = wav_scp[wav_id]
+            # check if it is really a pipe
+            if len(wav_original_pipe.split()) == 1:
+                wav_original_pipe = "cat {0} |".format(wav_original_pipe)
+            speech_dur = durations[wav_id]
+            max_noises_recording = math.floor(max_noises_per_minute * speech_dur / 60)
+            if prefix is not None:
+                new_wav_id = prefix + str(i) + "_" + wav_id
```

Comment:
@vijayaditya Sorry i don't get what you mean. What i need here is just the recording id, why do i need to inpu the spk_id ?